cmd/iop/shared.go

```diff
@@ -68,63 +69,78 @@ func genManifests(args *rootArgs) (name.ManifestMap, error) {
 		overlayYAML = string(b)
 	}
 
+	overlayFilenameLog := args.inFilename
+	if overlayFilenameLog == "" {
+		overlayFilenameLog = "[Empty Filename]"
+	}
+
 	// Start with unmarshaling and validating the user CR (which is an overlay on the base profile).
 	overlayICPS := &v1alpha2.IstioControlPlaneSpec{}
 	if err := util.UnmarshalWithJSONPB(overlayYAML, overlayICPS); err != nil {
+		log.Errorf("Could not unmarshal the overlay YAML from file: %s", overlayFilenameLog)
 		return nil, err
 	}
 	if errs := validate.CheckIstioControlPlaneSpec(overlayICPS, false); len(errs) != 0 {
+		log.Errorf("Overlay spec failed validation against IstioControlPlaneSpec: \n%v\n", overlayICPS)
```

Comment:
Do we need to log the whole overlayICPS value here, maybe just log fields that fail validation in validate function?

Reply:
Yes, especially when error happens. We want to know the internal state at that point.
+10 fields that fail validation should be logged, and it is. I will paste all sample logs in the PR description. Thanks.